### DISCUSSION TITLE
CSV parse: coerce String back to Array

### DISCUSSION THREAD

**Discussion by ADovgalyuk** [vectordotdev/vector#15776] [Q&A] [answered]
Hello everyone!
I have a csv file where some fields can be either empty or array. For example:
```
2022-12-23 13:01:16;111;['Main','Rule','Velocity'];['routerprotocol','rule','rule'];[2,1,0];[0,0,0];'','VelocityCheck',''];0;;3;-1.00;174;-1;tr-4;13;2;
```
I'm using function **parse_csv** that results in array with strings inside, but array fields from csv are also doublequoted as strings.
Code:
```
arr = parse_csv!(.message, delimiter: ";")
        .=compact(
            {
                "host" : "Test",
                "app" : "Test",
                "dateTime" : {{ arr[0]}},
                "Id" : {{ arr[1]}},
                "OneFieldWithArr" : {{ arr[2]}},
                "AnotherFieldWithArr" : {{ arr[3]}},
....
```
Result
```
{"app":"Protei_SS7FW","dateTime":"2022-12-23 13:01:16","host":"SS7FW","OneFieldWithArr":"['Main','Rule','Velocity']", "AnotherFieldWithArr":"['routerprotocol','rule','rule']" ......}
```

How can I correctly coerce value back to Array from String? 
Thanks for the help in advance!

**Comment by jszwedko** [comment, marked as answer]
Hey @ADovgalyuk !

Unfortunately I don't think there is a way to parse a string like that into an array in VRL right now. Would you want to open a [feature request](https://github.com/vectordotdev/vector/issues/new?assignees=&labels=type%3A+feature&template=feature.yml)?

As a workaround you can use the `lua` transform to do the parsing.
